packages/core/lib/SolidUtilContext.ts

```diff
@@ -1,25 +1,55 @@
-import type { IQueryEngine } from '@comunica/types';
-import type { QueryStringContext, QuerySourceContext } from '@rdfjs/types';
+import type {
+  AlgebraSparqlQueryable,
+  QueryAlgebraContext,
+  QuerySourceContext,
+  QueryStringContext,
+  SparqlResultSupport,
+  StringSparqlQueryable,
+} from '@rdfjs/types';
 import type { Session } from '@rubensworks/solid-client-authn-isomorphic';
+import type { Algebra } from 'sparqlalgebrajs';
 
-/**
- * A Solid utility context object contains data that can be used across different helper functions within SolidLabLib.
- */
-export interface SolidUtilContext {
+export interface IQueryStringContext<
+  SupportedResultType,
+  QueryContextType extends QueryStringContext & QuerySourceContext<string>
+> {
   /**
-   * An authenticated Solid session.
+   * A query engine object that can be used for finding data.
    */
-  session?: Session;
+  readonly engine: StringSparqlQueryable<SupportedResultType, QueryContextType>;
+  /**
+   * A context object for passing options to the query engine during query execution.
+   */
+  readonly queryContext: QueryContextType;
+}
+
+export interface IQueryAlgebraContext<
+  AlgebraType,
+  SupportedResultType,
+  QueryContextType extends QueryAlgebraContext & QuerySourceContext<string>
+> {
   /**
    * A query engine object that can be used for finding data.
    */
-  engine: IQueryEngine;
+  readonly engine: AlgebraSparqlQueryable<AlgebraType, SupportedResultType, QueryContextType>;
   /**
    * A context object for passing options to the query engine during query execution.
    */
-  queryContext: QueryStringContext & QuerySourceContext<string>;
+  readonly queryContext: QueryContextType;
 }
 
+/**
+ * A Solid utility context object contains data that can be used across different helper functions within SolidLabLib.
+ */
+export type SolidUtilContext =
+  IQueryAlgebraContext<Algebra.Operation, SparqlResultSupport, QueryAlgebraContext & QuerySourceContext<string>> &
+  IQueryStringContext<SparqlResultSupport, QueryStringContext & QuerySourceContext<string>> & {
+  /**
+   * An authenticated Solid session.
+   */
+    session?: Session;
+  };
+
```

Comment:
@rubensworks not after a full review yet as there is still more to port. I just want to know if you are on board with the typing changes in this file.

Reply:
Could you clarify a bit what the purpose of this PR is exactly?

I see a lot of query-related helpers being added in here.
Are you adding them just for the sake of having them? Or are they in function of new Solid-related packages that you'll be creating?
The helpers that I had added before are just here to enable the `getIdp()` functionality, but I have the feeling that the new helpers have no immediate purpose (yet?).

I just want the project to remain focused on its main goal, and avoid adding logic that would increase maintenance effort.
Unless of course we want to broaden the scope of this project, and also add a generic query toolkit.
Also looking at @RubenVerborgh for this.

Reply:
>  Are you adding them just for the sake of having them? Or are they in function of new Solid-related packages that you'll be creating?

These are functions that I've found myself re-writing for several use-cases (that are indeed query related) where the query engine is passed around in a config. These particular implementations are coming from a utility package in a query PoC that I'm aiming to make public this week. With that said, it probably does make sense to omit functions with hard-coded terms such as `queryLabel`.

> I just want the project to remain focused on its main goal ... unless of course we want to broaden the scope of this project, and also add a generic query toolkit.

I'm not entirely clear on what the full scope / goal of this project is indeed so it's is hard to judge exactly how much to port over here without that context.

Reply:
Quite happy to move these to an `@solidlab/query` package if that is more suitable; or close this PR if these types of functions don't belong here. @rubenverborgh do you have a position on whether or not this kind of thing is in scope?